src/data/finsupp/basic.lean

```diff
@@ -870,6 +870,13 @@ See `finsupp.lapply` for the stronger version as a linear map. -/
 @[simps apply]
 def apply_add_hom (a : α) : (α →₀ M) →+ M := ⟨λ g, g a, zero_apply, λ _ _, add_apply _ _ _⟩
 
+/-- Coercion from a `finsupp` to a function type is an `add_monoid_hom`. -/
+@[simps]
+noncomputable def coe_fn_add_monoid_hom : (α →₀ M) →+ (α → M) :=
```

Comment:
It would be good to make these names match. Note that `apply_add_hom` matches `single_add_hom`, so if you rename one you should rename them all.

The easy option is to call this `coe_fn_add_hom` to match what's already here, and leave someone else to change all the names later. I don't mind either way.